dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/EMQController.java

```diff
@@ -0,0 +1,42 @@
+package com.ibm.ws.collector.manager.buffer;
+
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import com.ibm.wsspi.collector.manager.BufferManager;
+
+public class EMQController {
+
+    protected final static ReentrantReadWriteLock RERWLOCK = new ReentrantReadWriteLock(true);
```

Comment:
I don't think u need a read-write lock for this. You can just synchronize on the list